scipy/stats/tests/test_distributions.py

```diff
@@ -2318,21 +2318,17 @@ def test_t_entropy():
 
 @pytest.mark.parametrize("methname", ["pdf", "logpdf", "cdf",
                                       "ppf", "sf", "isf"])
-def test_t_inf_df(methname):
-    t_dist = stats.t(df=np.inf, loc=3, scale=1)
+@pytest.mark.parametrize("df", [np.inf, [[np.inf, 1.], [2., np.inf]]])
+@pytest.mark.parametrize("x", [0.5, [0.5, 0.7]])
```

Comment:
Although it avoids indexing and makes the test stronger, I personally feel like the parameterization makes the test more complex to understand. Especially the line `np.diag(t_meth(x)) if np.ndim(df) else t_meth(x)` wouldn't work if the parameterization is changed.

> I'm getting a bit confused by the triple indexing

Oh, OK. I should have written some comments explaining the indexing. I will change both the tests to improve readability and to avoid confusion.

Reply:
I have pushed a commit to this branch. Does something like that work? If you don't agree, I will revert the change and also change the other test to use parameterization.

Reply:
> Although it avoids indexing and makes the test stronger, I personally feel like the parameterization makes the test more complex to understand

Sometimes I think that parameterizing tests makes them more complex. I don't think so here.  But since I feel the same way sometimes, it's fine how you prefer.

> wouldn't work if the parameterization is changed.

Depends on how you place the `np.inf`s. Had you placed them with a mask, you'd use that same mask to extract the results.

Reply:
With parameterization, masking makes sense. I will try that.

Reply:
Using masking with parametrization makes the tests concise and strong. Thanks for the suggestion! Let me know if this looks good to you. I will merge and continue working on the main PR.

Reply:
Yup looks good.